### ISSUE TITLE
GMT can't generate plots if username has whitespaces

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#2646]
See the screenshot below:
![image](https://user-images.githubusercontent.com/3974108/73663125-f9399680-466a-11ea-972e-44856963ec49.png)

**Comment by PaulWessel** [comment]
I can see the quoted filename is being passed in the call to psconvert.  However, in psconvert we strip the quotes from the file name via _noquote_name_.  We do add quotes back when calling gs, but I suspect some earlier step is calling _gmt_getdatapath_ (which gives all the "Look for file .." messages and clearly there we already have lost it.
Might be obvious if @joa-quim could replicate this in a dummy account with spaces in the name.

**Comment by seisman** [comment]
@PaulWessel You can also replicate it by creating a directory and setting it to HOME, e.g.
```
mkdir WHITE\ SPACE
cd WHITE\ SPACE
export HOME=`pwd`
gmt basemap -R-10/10/-10/5/0/10 -JX10c/3c -Baf -pdf map -Vd
```

**Comment by PaulWessel** [comment]
No, ran fine for me.  Temp files still written to /Users/pwessel/.gmt/sessions.  Maybe HOME not working for macOS?

**Comment by joa-quim** [comment]
Weird again. From *cmd* it errors but not clear why
```
gmt [DEBUG]: Shared Library # 1 (supplements). Path = C:/programs/GMTdir/gmt_plugins/supplements_w64.dll
gmt [DEBUG]: Shared Library # 2 (mbgmt). Path = C:/programs/GMTdir/gmt_plugins/mbgmt_w64.dll
begin [INFORMATION]: Creating a workflow directory "C:/v/spa co"/.gmt/sessions/gmt6.40852
gmt_mkdir (intermediate) error: Invalid argument
begin [ERROR]: Unable to create a workflow directory : "C:/v/spa co"/.gmt/sessions/gmt6.40852
```
trying to debug it from Julia, the firs step worked. Only the ``psconvert`` errored for the same reason.
```
julia> gmtend()
psconvert [ERROR]: Error for input file: No such file ('c:/v/spa)
```

**Comment by joa-quim** [comment]
The problem is that ``gmt_mkdir()`` (and probably others) aren't able to deal with paths that start with a ``"`` and ``API->session_dir`` has them. Tried to get rid of it but id I do it in API->session_dir in ``gmt_manage_workflow()`` then I have a crash at the end. 
Basically, I think we should never have double quotes in our internal string, only when passing to external programs so they know about this f... insistence in putting spaces in names. Note how shit propagates.

```
basemap [DEBUG]: Unable to create file "C:/v/spa co"/.gmt/sessions/gmt6.40852/gmt.frame
```

**Comment by PaulWessel** [comment]
Should be able to have a function that sanitizes paths before passing to system functions.  It should be easy to turn `"C:/v/spa co"/.gmt/sessions/gmt6.40852/gmt.frame` into `C:/v/spa co/.gmt/sessions/gmt6.40852/gmt.frame` inside gmt_mkdir gmt_fopen, gmt_access, etc.

**Comment by joa-quim** [comment]
Not yet good

```
gmt [DEBUG]: Shared Library # 2 (mbgmt). Path = C:/programs/GMTdir/gmt_plugins/mbgmt_w64.dll
begin [INFORMATION]: Creating a workflow directory "C:/v/spa co"/.gmt/sessions/gmt6.40852
gmt_mkdir (intermediate) error: Invalid argument
begin [ERROR]: Unable to create a workflow directory : "C:/v/spa co"/.gmt/sessions/gmt6.40852
gmt [ERROR]: Unable to call module begin from set_modern_mode_if_oneliner.
gmt [DEBUG]: Entering GMT_Destroy_Session
```

**Comment by seisman** [comment]
Just FYI, I could reproduce the issue on my macOS, and it was fixed by Paul's #2656 (at least for macOS).

**Comment by joa-quim** [comment]
Not on Win, and I don't see what is OS specific here
```
$ gmt basemap -R-10/10/-10/5/0/10 -JX10c/3c -Baf -Vd -pdf map -Vl
begin [INFORMATION]: Creating a workflow directory "C:/v/spa co"/.gmt/sessions/gmt6.40852
gmt_mkdir (intermediate) error: Invalid argument
begin [ERROR]: Unable to create a workflow directory : "C:/v/spa co"/.gmt/sessions/gmt6.40852
gmt [ERROR]: Unable to call module begin from set_modern_mode_if_oneliner.
```

**Comment by seisman** [comment]
This is what I see on macOS:
```
begin [INFORMATION]: Creating a workflow directory /Users/seisman/Desktop/workspace/SP ACE/.gmt/sessions/gmt6.44495
begin [DEBUG]: The PROJ_GEODESIC set to Vincenty
begin [DEBUG]: Look for file /Users/seisman/Desktop/workspace/SP ACE/gmt.conf
begin [DEBUG]: Look for file /Users/seisman/Desktop/workspace/SP ACE/.gmt/gmt.conf
begin [DEBUG]: Look for file /Users/seisman/Desktop/workspace/SP ACE/.gmt/server/gmt.conf
begin [DEBUG]: Look for file /Users/seisman/Desktop/workspace/SP ACE/.gmt/cache/gmt.conf
begin [DEBUG]: Could not find file gmt.conf
begin [DEBUG]: Set session name to be map pdf
begin [DEBUG]: Begin Workflow.  Session ID = 44495. Directory /Users/seisman/Desktop/workspace/SP ACE/.gmt/sessions/gmt6.44495 created.
begin [DEBUG]: GMT now running in modern mode [Session ID = 44495]
gmt [DEBUG]: Use PS filename /Users/seisman/Desktop/workspace/SP ACE/.gmt/sessions/gmt6.44495/gmt_0.ps-
```

**Comment by PaulWessel** [comment]
I will have a look, the 
`begin [INFORMATION]: Creating a workflow directory "C:/v/spa co"/.gmt/sessions/gmt6.40852`
message is a good start for me to look at.

**Comment by joa-quim** [comment]
That problem is simple. As I mentioned before ``gmt_mkdir()`` cannot cope with strings starting with a double quote. The true issue is why are double quotes still there.

**Comment by PaulWessel** [comment]
Closed as fixed by #2665.
